plugin/storage/badger/spanstore/rw_internal_test.go

```diff
@@ -100,48 +98,34 @@ func TestDecodeErrorReturns(t *testing.T) {
 	assert.Error(t, err)
 }
 
-func TestSortMergeIdsDuplicateDetection(t *testing.T) {
-	// Different IndexSeeks return the same results
-	ids := make([][][]byte, 2)
-	ids[0] = make([][]byte, 1)
-	ids[1] = make([][]byte, 1)
-	buf := new(bytes.Buffer)
-	binary.Write(buf, binary.BigEndian, uint64(0))
-	binary.Write(buf, binary.BigEndian, uint64(156697987635))
-	b := buf.Bytes()
-	ids[0][0] = b
-	ids[1][0] = b
-
-	query := &spanstore.TraceQueryParameters{
-		NumTraces: 64,
-	}
-
-	traces := sortMergeIds(query, ids)
-	assert.Equal(t, 1, len(traces))
-}
-
 func TestDuplicateTraceIDDetection(t *testing.T) {
 	runWithBadger(t, func(store *badger.DB, t *testing.T) {
 		testSpan := createDummySpan()
 		cache := NewCacheStore(store, time.Duration(1*time.Hour), true)
 		sw := NewSpanWriter(store, cache, time.Duration(1*time.Hour), nil)
 		rw := NewTraceReader(store, cache)
-
-		for i := 0; i < 8; i++ {
-			testSpan.SpanID = model.SpanID(i)
-			testSpan.StartTime = testSpan.StartTime.Add(time.Millisecond)
-			err := sw.WriteSpan(&testSpan)
-			assert.NoError(t, err)
+		origStartTime := testSpan.StartTime
+
+		traceCount := 128
+		for k := 0; k < traceCount; k++ {
+			testSpan.TraceID.Low = rand.Uint64()
+			for i := 0; i < 32; i++ {
+				testSpan.SpanID = model.SpanID(rand.Uint64())
+				testSpan.StartTime = origStartTime.Add(time.Duration(rand.Int31n(8000)) * time.Millisecond)
+				err := sw.WriteSpan(&testSpan)
+				assert.NoError(t, err)
+			}
 		}
 
 		traces, err := rw.FindTraceIDs(context.Background(), &spanstore.TraceQueryParameters{
 			ServiceName:  "service",
+			NumTraces:    128,
```

Comment:
Could we have similar tests where `NumTraces` is less than number of traces created - and then ensure that the most recent are returned.

Also might be worth adding an extra test that also adds a couple of tags into the mix, still ensuring that the `NumTraces` is less than the set of traces that would actually match, and ensure they are the most recent.

Reply:
Such test resides in the "TestIndexSeek". This "NumTraces" parameter is actually incorrect here, the intention is just to test that since 128*32 spans were created, only 128 is returned.